### DISCUSSION TITLE
Expose code fence attributes to SyntaxHighlighterAdapter::write_highlighted

### DISCUSSION THREAD

**Discussion by hongquan** [kivikakk/comrak#473] [Feature requests]
I'm building my personal blog, which let admin to write content in Markdown. The blog is about programming so highliting code block is important. But because the Comrak + Syntect combo is not strong enough, I want to create a flexible SyntaxHighlighterAdapter which can highlight code block with Syntect, or defer to client side, where I use [Shiki](https://shiki.style). To let this adapter know which side (server or client) to do the job, will add some extra attribute to code fence, like if it encounters:

```md

  ````rust {lines: true}
  const a: u8 = 1;
  ````\
```

It will let client side to do. But to do that, I need `SyntaxHighlighterAdapter::write_highlighted` to be passed the `attributes: HashMap<String, String>` like `write_code_tag` and `write_pre_tag`.

Could the API be changed?

**Comment by digitalmoksha** [comment]
@hongquan I probably misunderstand what you're trying to do. But if you're using your own syntax highlighter, then you don't need to use the Syntect component at all. That is what we do - we use a Ruby syntax highlighter and enable the `--full-info-string` option in order for the full info string (which would contain the `{lines: true}` from your example) to be passed through.

**Reply by hongquan** [reply]
@digitalmoksha I want both, as in previous comment:

> I want to create a *flexible* SyntaxHighlighterAdapter which can highlight code block with Syntect, *or* defer to client side.

Most of the cases, I want to highlight at backend side, only some cases I defer to client side (JS).
